### PR TITLE
[PyOV] Add README.md and align structure-showing style

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Improving developer documentation.
 - Removal of old `README.md` from compatibility folder.
 
### Tickets:
 - *81826*

### CODE REVIEW

File: src/bindings/python/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on Github.
+
+Team members and main responsibilities:
+
+* [Jan Iwaszkiewicz](https://github.com/jiwaszki) (e-mail: jan.iwaszkiewicz@intel.com)  
+Techincal Lead: questions, issues and feature requests, memory and GIL related questions
+* [Anastasia Kuporosova](https://github.com/akuporos) (e-mail: anastasia.kuporosova@intel.com)  
+Developer: building of the component, general bindings
+* [Przemyslaw Wysocki](https://github.com/p-wysocki) (e-mail: przemyslaw.wysocki@intel.com)  
+Developer: project requirements, general bindings
```
Comment (ilyachur) [inline src/bindings/python/README.md:23]: Please use github python maintainer instead of list of persons

File: src/bindings/python/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# OpenVINO Python API
+
+OpenVINO Python API is a part of OpenVINO library. The component is responsible for:
+
+* Bindings of OpenVINO - allowing users to utilize OpenVINO library in their Python code. Python API provides bindings to basic and advanced APIs from OpenVINO Runtime.
+* Extending OpenVINO with pythonic features - on top of direct translations from C++, Python API component:
+    * Adds specific extensions to support numpy-based data.
+    * Provides support for external frameworks inputs.
+    * Provides shortcuts and helpers with more pythonic design.
+    * Allows to apply advanced concepts like shared memory to take full advantage of OpenVINO.
```
Comment (ilyachur) [inline src/bindings/python/README.md:10]: Do we have any code styles and something like that for python?

File: src/bindings/python/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Techincal Lead: questions, issues and feature requests, memory and GIL related questions
+* [Anastasia Kuporosova](https://github.com/akuporos) (e-mail: anastasia.kuporosova@intel.com)  
+Developer: building of the component, general bindings
+* [Przemyslaw Wysocki](https://github.com/p-wysocki) (e-mail: przemyslaw.wysocki@intel.com)  
+Developer: project requirements, general bindings
+
+## Components
+
+```
+openvino/src/bindings/python
```
Comment (ilyachur) [inline src/bindings/python/README.md:28]: We already exists here :)
Maybe we can use a usual list as we have for other components?

File: src/bindings/python/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+* Bindings of OpenVINO - allowing users to utilize OpenVINO library in their Python code. Python API provides bindings to basic and advanced APIs from OpenVINO Runtime.
+* Extending OpenVINO with pythonic features - on top of direct translations from C++, Python API component:
+    * Adds specific extensions to support numpy-based data.
+    * Provides support for external frameworks inputs.
+    * Provides shortcuts and helpers with more pythonic design.
+    * Allows to apply advanced concepts like shared memory to take full advantage of OpenVINO.
+
+## Key contacts
+
+If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on Github.
```
Comment (p-wysocki) [inline src/bindings/python/README.md:14]:
```suggestion
If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on GitHub.
```

File: src/bindings/python/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+├── thirdparty              <-- Thirdparty libraries
+│   └── pybind11
+└── wheel                   <-- Wheel specific directory
+```
+
+## Tutorials
+
+If you want to contribute to OpenVINO Python API, here is the list of learning materials and project's guidelines:
+
+* [How to contribute](docs/contributing.md)
```
Comment (ilyachur) [inline src/bindings/python/README.md:47]: Do we have a plan to change and review these files?
At the current moment it looks strange:
Screenshot 2022-11-18 at 16 24 46

File: src/bindings/python/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+├── tests
+├── tests_compatibility
+├── thirdparty              <-- Thirdparty libraries
+│   └── pybind11
+└── wheel                   <-- Wheel specific directory
+```
+
+## Tutorials
+
+If you want to contribute to OpenVINO Python API, here is the list of learning materials and project's guidelines:
```
Comment (p-wysocki) [inline src/bindings/python/README.md:45]:
```suggestion
If you want to contribute to OpenVINO Python API, here is the list of learning materials and project guidelines:
```

**Review by p-wysocki — APPROVED** [review]

File: src/bindings/python/docs/code_examples.md [hunk 7]
```diff
@@ -61,18 +61,18 @@ But how to extend existing API? Let's navigate to `openvino/src/bindings/python/
```
Comment (ilyachur) [inline src/bindings/python/docs/code_examples.md:61]: I would like to review these files. For example why here: `openvino/src/bindings/python/src/openvino` we don't have a link? :)

File: src/bindings/python/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+* [How to contribute](docs/contributing.md)
+* [How to extend OpenVINO Python API](docs/code_examples.md)
+* [How to test OpenVINO Python API](docs/test_examples.md)
+* [How to upgrade local Python version](docs/python_version_upgrade.md)
+
+## See also
+
+Related pages and articles:
+* [pybind11 repository](https://github.com/pybind/pybind11)
```
Comment (ilyachur) [inline src/bindings/python/README.md:55]: I propose to add links on high level documents as we have for other components.

File: src/bindings/python/README.md (hunk 6 above)
Comment (jiwaszki) [reply src/bindings/python/README.md]: Done

File: src/bindings/python/README.md (hunk 4 above)
Comment (jiwaszki) [reply src/bindings/python/README.md]: Got it!

File: src/bindings/python/README.md (hunk 2 above)
Comment (jiwaszki) [reply src/bindings/python/README.md:10]: I will add that if necessary. However we are following adjusted style checks and everything is described in [contributing.md](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/contributing.md), so link against it is enough?

File: src/bindings/python/docs/code_examples.md (hunk 7 above)
Comment (jiwaszki) [reply src/bindings/python/docs/code_examples.md:61]: Feel free to do so, let me offline on your review. You can follow https://github.com/openvinotoolkit/openvino/pull/12145 for additional context.

**Comment by akuporos** [comment]
Let's clean up also this file: https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/compatibility/openvino/README.md

**Comment by akuporos** [comment]
May be it worth to add a link to https://docs.openvino.ai/latest/api/api_reference.html?
CC: @ilyachur

**Comment by jiwaszki** [comment]
> May be it worth to add a link to https://docs.openvino.ai/latest/api/api_reference.html? CC: @ilyachur

It is already added in section called `See also`.

File: src/bindings/python/README.md (hunk 3 above)
Comment (jiwaszki) [reply src/bindings/python/README.md:19]: I see both options as equal for developers. However, if it is general practice across components, sure I will edit this.

File: src/bindings/python/README.md (hunk 2 above)
Comment (jiwaszki) [reply src/bindings/python/README.md:10]: Linked with current state.

File: src/bindings/python/README.md (hunk 3 above)
Comment (jiwaszki) [reply src/bindings/python/README.md]: Done

**Review by akuporos — APPROVED** [review]

File: src/bindings/python/README.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    * Adds specific extensions to support numpy-based data.
+    * Provides support for external frameworks inputs.
+    * Provides shortcuts and helpers with more pythonic design.
+    * Allows to apply advanced concepts like shared memory to take full advantage of OpenVINO.
+
+OpenVINO Python API uses [common codestyle checks](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/contributing.md#contribution-guidelines-and-best-practices) that are adjusted to project needs.
+
+## Key contacts
+
+If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on GitHub via [openvino-ie-python-api-maintainers](https://github.com/orgs/openvinotoolkit/teams/openvino-ie-python-api-maintainers). You can always directly contact everyone from this group.
```
Comment (ilyachur) [inline src/bindings/python/README.md:16]: nitpick: If I remember right you cannot write a message on github group :)

**Review by ilyachur — APPROVED** [review]
@tsavina Please take a look

File: src/bindings/python/README.md (hunk 9 above)
Comment (jiwaszki) [reply src/bindings/python/README.md:16]: Most of us have corporate emails shown in their profiles, thus I pointed to direct contact as an option.

File: src/bindings/python/README.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
+# OpenVINO Python API
+
+OpenVINO Python API is a part of OpenVINO library. The component is responsible for:
```
Comment (tsavina) [inline src/bindings/python/README.md:3]:
```suggestion
OpenVINO Python API is a part of the OpenVINO library. The component is responsible for:
```

File: src/bindings/python/README.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+* Bindings of OpenVINO - allowing users to utilize OpenVINO library in their Python code. Python API provides bindings to basic and advanced APIs from OpenVINO Runtime.
+* Extending OpenVINO with pythonic features - on top of direct translations from C++, Python API component:
+    * Adds specific extensions to support numpy-based data.
+    * Provides support for external frameworks inputs.
+    * Provides shortcuts and helpers with more pythonic design.
+    * Allows to apply advanced concepts like shared memory to take full advantage of OpenVINO.
+
+OpenVINO Python API uses [common codestyle checks](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/contributing.md#contribution-guidelines-and-best-practices) that are adjusted to project needs.
```
Comment (tsavina) [inline src/bindings/python/README.md:12]:
```suggestion
OpenVINO Python API uses [the common codestyle checks](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/contributing.md#contribution-guidelines-and-best-practices) which are adjusted to project needs.
```

File: src/bindings/python/README.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+OpenVINO PYTHON API has the following structure:
+
+* [docs](./docs/) - folder that contains developer documentation and code examples.
+* [src](./src/) - folder with all source files for Python API.
+    * [src/compatibility](./src/compatibility/) - sources for compatibility API, including older modules like `ngraph` and `openvino.inference_engine`.
+    * [src/compatibility](./src/openvino/) - Python sources for current API.
+    * [src/pyopenvino](./src/pyopenvino/) - C++ sources for current API.
+* [tests](./tests/) - tests directory for current API.
+* [tests_compatibility](./tests_compatibility/) - tests directory for compatibility API.
+* [thirdparty](./thirdparty/) - folder that contains thirdparty modules like `pybind11`.
```
Comment (tsavina) [inline src/bindings/python/README.md:29]:
```suggestion
* [thirdparty](./thirdparty/) - folder that contains third-party modules like `pybind11`.
```

File: src/bindings/python/README.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+* [wheel](./wheel/) - wheel specific directory, contains all specifc requirements and files used during wheel creation.
```
Comment (tsavina) [inline src/bindings/python/README.md:30]:
```suggestion
* [wheel](./wheel/) - wheel-specific directory that contains all specific requirements and files used during wheel creation.
```

File: src/bindings/python/README.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+If you have any questions, feature requests or want us to review your PRs, send us a message or ping us on GitHub via [openvino-ie-python-api-maintainers](https://github.com/orgs/openvinotoolkit/teams/openvino-ie-python-api-maintainers). You can always directly contact everyone from this group.
+
+## Components
+
+OpenVINO PYTHON API has the following structure:
+
+* [docs](./docs/) - folder that contains developer documentation and code examples.
+* [src](./src/) - folder with all source files for Python API.
+    * [src/compatibility](./src/compatibility/) - sources for compatibility API, including older modules like `ngraph` and `openvino.inference_engine`.
+    * [src/compatibility](./src/openvino/) - Python sources for current API.
```
Comment (tsavina) [inline src/bindings/python/README.md:25]: the directories have the same name but the links are different, need to change the second one?

Comment (jiwaszki) [reply src/bindings/python/README.md:25]: Great catch!

**Pull request merged**